src/core/issues-notes.js

```diff
@@ -139,28 +139,28 @@ export function run(conf, doc, cb) {
   var $ins = $(".issue, .note, .warning, .ednote"),
     ghIssues = {},
     issueBase = conf.issueBase;
+  async function asyncFetch() {
```

Comment:
I'm not a huge fan of relying on the variables in outer scope, tbh. Can we move `async` up whatever contains this function?  The we won't need `asyncFetch` at all.

Reply:
we are using  ``` if ( $ins.length ) ``` that is in the ``` run ``` and then ``` asyncFetch ``` is called.
so essentially we are just invoking the funcion inside if condition, how is ``` asyncfetch ``` relying on the global variables?

Am I missing something, correct me if i'm wrong.

I didn't get your solution.

Reply:
> relying on the global variables?

Sorry, bad terminology on my part... "global" in the sense of the function scope. That can get confusing, specially when things change async. 

to be clear, we don't want things like this:

```JS
function run(){
   var x;
   async function y(){
        x++;
   }
   x = 1;
   y().then(somethingElse);
}
```

We want:

```JS
async function run(){
   var x;
   x = 1;
   x++;
   await y();
   await somethingElse();
}
```

Reply:
couple of fixes above... hopefully that's more clear.

Reply:
got it, made the changes